core/trie2/triedb/pathdb/buffer.go

```diff
@@ -0,0 +1,75 @@
+package pathdb
+
+import (
+	"fmt"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/core/trie2/trienode"
+	"github.com/NethermindEth/juno/core/trie2/trieutils"
+	"github.com/NethermindEth/juno/db"
+)
+
+// Stores the pending nodes in memory to be committed later
+type buffer struct {
+	layers uint64
+	limit  uint64
+	nodes  *nodeSet
+}
+
+func newBuffer(limit int, nodes *nodeSet, layer uint64) *buffer {
+	if nodes == nil {
+		nodes = newNodeSet(nil, nil, nil)
+	}
+	return &buffer{
+		limit:  uint64(limit),
+		nodes:  nodes,
+		layers: layer,
+	}
+}
+
+func (b *buffer) node(owner *felt.Felt, path trieutils.Path, isClass bool) (trienode.TrieNode, bool) {
+	return b.nodes.node(owner, path, isClass)
+}
+
+func (b *buffer) commit(nodes *nodeSet) *buffer {
+	b.layers++
+	b.nodes.merge(nodes)
+	return b
+}
+
+func (b *buffer) reset() {
+	b.layers = 0
+	b.limit = 0
+	b.nodes.reset()
+}
+
+func (b *buffer) isFull() bool {
+	return b.nodes.size > b.limit
+}
+
+func (b *buffer) flush(d db.KeyValueStore, cleans *cleanCache, id uint64) error {
```

Comment:
Please change the name `d`,  maybe use `kvs` or  `kvStore` but `d` is not very good in my opinion